test/OrchardCore.Tests/Settings/SiteExtensionsTests.cs

```diff
@@ -0,0 +1,209 @@
+        {
+            var localizationStartup = new OrchardCoreLocalization.Startup();
+            var webHostBuilder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                    SetupSiteSettingsCultures(defaultCulture, supportedCultures);
+                    services.AddTransient<ISiteService>(_ => new SiteService(_site.Object));
+
+                    services.AddRouting();
+
+                    localizationStartup.ConfigureServices(services);
+                })
+                .Configure(app =>
+                {
+                    localizationStartup.Configure(app, new RouteBuilder(app), app.ApplicationServices);
+
+                    app.Run(context =>
+                    {
+                        var requestLocalizationOptions = context.RequestServices.GetService<IOptions<RequestLocalizationOptions>>().Value;
+
+                        Assert.Equal(expectedDefaultCulture, requestLocalizationOptions.DefaultRequestCulture.Culture.Name);
+                        Assert.Equal(expectedSupportedCultures, requestLocalizationOptions.SupportedCultures.Select(c => c.Name).ToArray());
+
+                        return Task.FromResult(0);
+                    });
+                });
+
+            using (var server = new TestServer(webHostBuilder))
+            {
+                var client = server.CreateClient();
+                var requestedCulture = "en";
+                var response = await client.GetAsync($"page/?culture={requestedCulture}&ui-culture={requestedCulture}");
+            }
+        }
+
+        private async Task RunTestWithAcceptLanguageHttpHeader(string acceptLanguages, string defaultCulture, string[] supportedCultures, string expectedDefaultCulture, string[] expectedSupportedCultures)
+        {
+            var localizationStartup = new OrchardCoreLocalization.Startup();
+            var webHostBuilder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                    SetupSiteSettingsCultures(defaultCulture, supportedCultures);
+                    services.AddTransient<ISiteService>(_ => new SiteService(_site.Object));
+
+                    services.AddRouting();
+
+                    localizationStartup.ConfigureServices(services);
+                })
+                .Configure(app =>
+                {
+                    localizationStartup.Configure(app, new RouteBuilder(app), app.ApplicationServices);
+
+                    app.Run(context =>
+                    {
+                        var requestLocalizationOptions = context.RequestServices.GetService<IOptions<RequestLocalizationOptions>>().Value;
+
+                        Assert.Equal(expectedDefaultCulture, requestLocalizationOptions.DefaultRequestCulture.Culture.Name);
+                        Assert.Equal(expectedSupportedCultures, requestLocalizationOptions.SupportedCultures.Select(c => c.Name).ToArray());
```

Comment:
This of course will never fail because there is no InlineData that tests when a default culture is different than the supported cultures.

Reply:
Is this mean that default culture is not in the supported culture list? If it's YES, i'm not sure why we need to test this, IMHO testing the result of the localization middleware is enough to ensure that everything works smoothly .. please correct me if I'm wrong

Reply:
Here to make the asserts usefull you would need to have the defaultculture in the expectedSupportedCultures. Else nothing alters your values that are proposed in your InlineData. The expectedSupportedCultures should be variable based on the user system culture. Here we need to make some adjustments to test the proper values.